File: lotka_volterra/lotka_volterra.jl
using OrdinaryDiffEq

function system(du, u, p, t)
    x, y = u
    α, β, δ, γ = p
    du[1] = α * x - β * x * y
    du[2] = δ * x * y - γ * y
end

function generate_dataset(num_simulations, t_span, params)
    tspan = (t_span[1], t_span[2])
    t = range(tspan[1], tspan[2], length=200)
    
    initial_conditions = rand(num_simulations, 2) .* 10
    dataset = []
    
    for ic in eachrow(initial_conditions)
        prob = ODEProblem(system, ic, tspan, params)
        sol = solve(prob, Tsit5(), saveat=t)
        push!(dataset, Array(sol))
    end
    
    return cat(dataset..., dims=3), collect(t)
end

File: lotka_volterra/lotka_volterra.py
import numpy as np
from scipy.integrate import odeint

def system(state, t, alpha, beta, delta, gamma):
    x, y = state
    dxdt = alpha * x - beta * x * y
    dydt = delta * x * y - gamma * y
    return [dxdt, dydt]

def simulate(initial_conditions, t, params):
    alpha, beta, delta, gamma = params
    solutions = []
    for ic in initial_conditions:
        solution = odeint(system, ic, t, args=(alpha, beta, delta, gamma))
        solutions.append(solution)
    return np.array(solutions)

def generate_dataset(num_simulations, t_span, params):
    np.random.seed(42)
    t = np.linspace(t_span[0], t_span[1], 200)
    initial_conditions = np.random.rand(num_simulations, 2) * 10
    dataset = simulate(initial_conditions, t, params)
    return dataset, t